src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2454,7 +2454,11 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
 
     # for each addons argument
     for addon_arg in addon_args:
-        addon = ADDONS[addon_arg]
+        if addon_arg in ADDONS:
+            addon = ADDONS[addon_arg]
+        else:
+            raise KeyError(f"Key {addon_arg} not in ADDONS dict.")
```

Comment:
assuming this is a generic error if an addon name is passed which does not exist, suggest the following error if we can make a best guess suggestion and output of the known list.

"The addon <name> is not a recognized addon option. Did you mean <working addon name>? Valid options include <addon list>."